### PR TITLE
Proposed 1.3 release

### PR DESCRIPTION
**Pull request by tomzo**
* rewritten chef resources to use new custom resource syntax
 * updated agent startup config files to use `GO_SERVER_URL` instead of host and port
 * by default install java 8
 * isolated `gocd_agent_autoregister_file` resource
 * added golang agent support
 * use apt cookbook >= 3.0

I'll be testing this in next days before releasing. 
Any feedback welcome.

### CODE REVIEW

**Comment by ketan** [comment]
FYI. We have turned on the http protocol so there is no need for apt-protocol-https anymore. As long as you are importing the GPG keys properly, the packages can be guaranteed to be untempered.

File: README.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ If you have been using go-cookbook previously then please note that:
 
 ### Java
 
-Please note that java (>= 7) is needed to run Go server and agents. This cookbook
+Please note that java 8 is recommended to run Go server and agents. This cookbook
```
Comment (philoserf) [inline README.md]: Java 7 is deprecated already and will not be supported in GoCD 17.

https://www.go.cd/releases/

**Comment by philoserf** [comment]
Results of the Rakefile style check:

```plain
Starting Foodcritic linting...
FC009: Resource attribute not recognised: go-cookbook/recipes/agent_windows_install.rb:21
FC009: Resource attribute not recognised: go-cookbook/recipes/server_windows_install.rb:13
FC019: Access node attributes in a consistent manner: go-cookbook/recipes/java.rb:2
```

**Comment by philoserf** [comment]
Results of Rakefile unit:chefspec check:

```plain
Finished in 29.99 seconds (files took 1.66 seconds to load)
155 examples, 0 failures


ChefSpec Coverage report generated...

  Total Resources:   59
  Touched Resources: 45
  Touch Coverage:    76.27%

Untouched Resources:

  ohai[reload_passwd_for_go_user]    gocd/recipes/agent_linux_install.rb:3
  apt_package[apt-transport-https]   gocd/recipes/repository.rb:4
  directory[/mnt/big_drive]          gocd/providers/agent.rb:9
  directory[/var/log/my-go-agent]    gocd/providers/agent.rb:9
  directory[/mnt/big_drive/config]   gocd/providers/agent.rb:15
  bash[setup init.d for my-go-agent]   gocd/providers/agent.rb:21
  link[/usr/share/my-go-agent]       gocd/providers/agent.rb:29
  bash[setup init.d for go-agent]    gocd/providers/agent.rb:21
  link[/usr/share/go-agent]          gocd/providers/agent.rb:29
  ohai[reload_passwd]                gocd/recipes/server_linux_install.rb:3
  apt_package[unzip]                 gocd/recipes/server_linux_install.rb:9
  http_request[verify_go-server_comes_up]   gocd/recipes/server.rb:16
  ruby_block[publish_autoregister_key]   gocd/recipes/server.rb:23
  yum_package[unzip]                 gocd/recipes/server_linux_install.rb:9
```

**Pull request merged**